app/helpers/sessions_helper.rb

```diff
@@ -38,7 +39,30 @@ def clear_cart
     end
     
     def log_out
+        user = User.find_by(id: session[:user_id])
+        user.update_attribute(:cart, hto_s(session[:cart]))
```

Comment:
hash.to_s works, but it doesn't output a string with the format I'm looking
for. hto_s and sto_h are only to be used with a cart. Since a cart has the
format {item_id => quantity}, we needn't worry about having nested hashes.
Thanks for the heads up on update_attribute though, I will read up on that.

On Tue, Jun 2, 2015 at 12:22 AM Kevin Chen <notifications@github.com> wrote:

> In app/helpers/sessions_helper.rb
> <https://github.com/Niupiao/niupiao-market/pull/31#discussion_r31492149>:
>
> > @@ -38,7 +39,30 @@ def clear_cart
> >      end
> >
> >      def log_out
> > +        user = User.find_by(id: session[:user_id])
> > +        user.update_attribute(:cart, hto_s(session[:cart]))
>
> @modsoussi <https://github.com/modsoussi> Note that update_attribute
> skips validations (see handy blog post on updating models
> <http://www.davidverhasselt.com/set-attributes-in-activerecord/>).
>
> —
> Reply to this email directly or view it on GitHub
> <https://github.com/Niupiao/niupiao-market/pull/31/files#r31492149>.
>